src/Controller/Traits/UserValidationTrait.php

```diff
@@ -40,7 +41,14 @@ public function validate($type = null, $token = null)
                 case 'email':
                     try {
                         $result = $this->getUsersTable()->validate($token, 'activateUser');
+
                         if ($result) {
+                            $event = $this->dispatchEvent(UsersAuthComponent::EVENT_AFTER_REGISTER_CONFIRM, [
+                                'user' => $result
+                            ]);
+                            if ($event->result instanceof Response) {
+                                return $this->redirect(['action' => 'login']);
```

Comment:
This condition could be improved to return the response object like in https://github.com/CakeDC/users/blob/master/src/Controller/Traits/RegisterTrait.php#L137

Reply:
Please make to check it with unit test. Thank you.